source/speech.py

```diff
@@ -1939,3 +2008,489 @@ def __repr__(self):
 		if self.text:
 			out += ", text=%r" % self.text
 		return out + ")"
+
+class BaseCallbackCommand(SpeechCommand):
+	"""Base class for commands which cause a function to be called when speech reaches them.
+	This class should not be instantiated directly.
+	It is designed to be subclassed to provide specific functionality;
+	e.g. L{BeepCommand}.
+	To supply a generic function to run, use L{CallbackCommand}.
+	This command is never passed to synth drivers.
+	"""
+
+	def run(self):

```

Comment:
This looks like an abstract method. Could you mark it as such by adding the abc.ABCMeta meta class and decorating it with abc.abstractmethod?

Reply:
Ugh, just realised that this is going to fail when setting run in __init__, as happens in the CallbackCommand. If you really want to mark run as abstract, you will probably have to save the callback as _run instead and then make run execute _run. This is not very elegant. I leave it up to you whether you want to use the abc framework here, or just leave it as is.